includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,553 @@
+			CLASSIFAI_PLUGIN_VERSION,
+			true
+		);
+
+		wp_localize_script(
+			'classifai-gutenberg-plugin',
+			'classifaiTextToSpeechData',
+			[
+				'supportedPostTypes' => \Classifai\get_supported_post_types_for_azure_text_to_speech(),
+				'noPermissions'      => ! is_user_logged_in() || ! current_user_can( 'edit_post', $post->ID ),
+			]
+		);
+	}
+
+	/**
+	 * Can the functionality be initialized?
+	 *
+	 * @return bool
+	 */
+	public function can_register() {
+		$options = get_option( $this->get_option_name() );
+
+		if ( empty( $options ) || ( isset( $options['authenticated'] ) && false === $options['authenticated'] ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Register the actions needed.
+	 */
+	public function register() {
+		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_editor_assets' ] );
+		add_filter( 'rest_api_init', [ $this, 'add_synthesize_speech_meta_to_rest_api' ] );
+		add_filter( 'the_content', [ $this, 'render_post_audio_controls' ] );
+	}
+
+	/**
+	 * Resets settings for the Personalizer provider.
+	 */
+	public function reset_settings() {
+		update_option( $this->get_option_name(), $this->get_default_settings() );
+	}
+
+	/**
+	 * Set up the fields for each section.
+	 */
+	public function setup_fields_sections() {
+		add_settings_section( $this->get_option_name(), $this->provider_service_name, '', $this->get_option_name() );
+		$default_settings = $this->get_default_settings();
+		$voices_options   = $this->get_voices_select_options();
+
+		add_settings_field(
+			'url',
+			esc_html__( 'Endpoint URL', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'url',
+				'input_type'    => 'text',
+				'default_value' => $default_settings['url'],
+				'description'   => __( 'Supported protocol and hostname endpoints, e.g., <code>https://REGION.api.cognitive.microsoft.com</code> or <code>https://EXAMPLE.cognitiveservices.azure.com</code>. This can look different based on your setting choices in Azure.', 'classifai' ),
+			]
+		);
+
+		add_settings_field(
+			'api-key',
+			esc_html__( 'API Key', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'api_key',
+				'input_type'    => 'password',
+				'default_value' => $default_settings['api_key'],
+			]
+		);
+
+		add_settings_field(
+			'post-types',
+			esc_html__( 'Post Types', 'classifai' ),
+			[ $this, 'render_post_types_checkboxes' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'option_index' => 'post_types',
+			]
+		);
+
+		if ( ! empty( $voices_options ) ) {
+			add_settings_field(
+				'voice',
+				esc_html__( 'Voice', 'classifai' ),
+				[ $this, 'render_select' ],
+				$this->get_option_name(),
+				$this->get_option_name(),
+				[
+					'label_for'     => 'voice',
+					'options'       => $voices_options,
```

Comment:
Currently, we are showing the list of all voices, could we limit it by language? Site in the English language should show only English voices. what are your thoughts on this?

Reply:
@iamdharmesh as discussed on Slack, we can leave this as it for now as it doesn't have a major impact.